Documentation/upgrade.md

```diff
@@ -0,0 +1,293 @@
+Only the one daemon set will need to be edited to update the image version, then each OSD pod will need to be deleted so that a new pod will be created by the daemon set to take its place.
+* **Specify individual nodes:** each storage node specified in the cluster spec will be managed by its own individual replica set.
+Each of these replica sets will need to be edited to update the image version, then each OSD pod will need to be deleted so its replica set will start a new pod on the new version to replace it.
+
+In this example, we are going to walk through the case where `useAllNodes: true` was set in the cluster spec, so there will be a single daemon set managing all the OSD pods.
+
+Let's edit the image version of the pod template in the daemon set:
+```bash
+kubectl -n rook edit daemonset rook-ceph-osd
+```
+
+The relevant snippet to update in your editor is shown updated to `rook/rook:v0.5.1` below:
+```
+        image: rook/rook:v0.5.1
+        imagePullPolicy: IfNotPresent
+        name: rook-ceph-osd
+```
+
+Once the daemon set is updated, we can begin deleting each OSD pod **one at a time** and verifying a new one comes up to replace it that is running the new version.
+After each pod, the cluster health and OSD status should remain or return to an okay state as described in the [health verification section](#health-verification).
+To get the names of all the OSD pods, the following can be used:
+```bash
+kubectl -n rook get pod -l app=rook-ceph-osd -o jsonpath='{range .items[*]}{.metadata.name}{"\n"}{end}'
+```
+
+Below is an example of deleting just one of the OSD pods (note that the names of your OSD pods will be different):
+```bash
+kubectl -n rook delete pod rook-ceph-osd-kcj8f
+```
+
+The status and version for all OSD pods can be collected with the following command and its sample output:
+```bash
+> kubectl -n rook get pod -l app=rook-ceph-osd -o jsonpath='{range .items[*]}{.metadata.name}{" "}{.status.phase}{" "}{.spec.containers[0].image}{"\n"}{end}'
+rook-ceph-osd-7fmdl Running rook/rook:v0.5.1
+rook-ceph-osd-ntfx0 Running rook/rook:v0.5.1
+```
+
+Remember after each OSD pod to verify the cluster health using the instructions found in the [health verification section](#health-verification).
+
+### Ceph Manager
+Similar to the Rook API and operator, the Ceph manager pods are managed by a deployment.
+We will edit the deployment to use the new image version of `rook/rook:v0.5.1`:
+```bash
+kubectl -n rook edit deployment rook-ceph-mgr0
+```
+
+Below is the relevant snippet to update in your editor to use the new `rook/rook:v0.5.1` image version:
+```
+        image: rook/rook:v0.5.1
+        imagePullPolicy: IfNotPresent
+        name: rook-ceph-mgr0
+```
+
+To verify that the manager pod is `Running` and on the new version, use the following:
+```bash
+kubectl -n rook get pod -l app=rook-ceph-mgr -o jsonpath='{range .items[*]}{.metadata.name}{" "}{.status.phase}{" "}{.spec.containers[0].image}{"\n"}{end}'
+```
+
```

Comment:
Should you also have sections to upgrade RGW and MDS?

Reply:
I thought about adding those but I left it as just the "core" components.  For completeness, I should add RGW/MDS also.